datafusion/core/tests/dataframe/mod.rs

```diff
@@ -2050,6 +2050,139 @@ async fn test_dataframe_placeholder_missing_param_values() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn test_dataframe_placeholder_column_parameter() -> Result<()> {
+    let ctx = SessionContext::new();
+
+    let df = ctx.read_empty().unwrap().select_exprs(&["$1"]).unwrap();
+
+    let logical_plan = df.logical_plan();
+    let formatted = logical_plan.display_indent_schema().to_string();
+    let actual: Vec<&str> = formatted.trim().lines().collect();
+    let expected = vec!["Projection: $1 [$1:Null;N]", "  EmptyRelation []"];
+    assert_eq!(
+        expected, actual,
+        "\n\nexpected:\n\n{expected:#?}\nactual:\n\n{actual:#?}\n\n"
+    );
+
+    // The placeholder is not replaced with a value,
+    // so the filter data type is not know, i.e. a = $0.
+    // Therefore, the optimization fails.
+    let optimized_plan = ctx.state().optimize(logical_plan);
+    assert!(optimized_plan.is_err());
```

Comment:
I think this line is redundant -- as `unwrap_err()` will panic if optimize_plan is not an err